diff/keys_test.go

```diff
@@ -65,14 +72,22 @@ func TestConfigurationWithAddedCompareConfigs(t *testing.T) {
 	}
 
 	g1, err := ucfg.NewFrom(oneGraph, opts...)
-	assert.NoError(t, err)
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	g2, err := ucfg.NewFrom(twoGraph, opts...)
-	assert.NoError(t, err)
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	d := CompareConfigs(g1, g2, opts...)
-	assert.True(t, d.HasChanged())
-	assert.True(t, d.HasKeyAdded())
+	if !d.HasChanged() {
+		t.Fatal("expected Diff.HashChanged() to be true")
+	}
```

Comment:
This reminds me that `assert`, and `require` package allow you to add custom messages. For example:
`assert.True(t, d.HasChanged(), "expected Diff.HasChanged() to be true")`.

All the if conditions make me wonder if we want more test helpers like `func expectChanged(t *testing.T, d Diff) { ... }`. Then the test becomes: `expectedChanged(t, CompareConfigs(g1, g2, opts...))`.

Reply:
It's deceptively repetitive.

`!d.HasChanged()` three times
`d.HasChanged()` once.
`!d.HasKeyRemoved()` once.
`d.HasKeyRemoved()` once.